src/com/goide/intentions/GoMoveToStructInitializationIntention.java

```diff
@@ -238,59 +293,62 @@ private static boolean isUninitializedFieldReferenceExpression(@Nullable GoRefer
   public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element) throws IncorrectOperationException {
     Data data = getData(element);
     if (data == null) return;
-    moveFieldReferenceExpressions(data);
+
+    boolean needReplaceDeclaration = data.myCompositeLit == null;
+    GoCompositeLit compositeLit = needReplaceDeclaration ? createStructLiteral(data.myStructDefinition, project) : data.myCompositeLit;
+    if (compositeLit == null) return;
+
+    moveFieldReferenceExpressions(data.myReferenceExpressions, compositeLit.getLiteralValue(), data.myAssignment);
+    if (needReplaceDeclaration) {
+      String definitionText = data.myStructDefinition.getText();
+      GoStatement shortVarStatement = GoElementFactory.createShortVarDeclarationStatement(project, definitionText, compositeLit.getText());
+      data.myStructDeclaration.replace(shortVarStatement);
+    }
   }
 
-  private static void moveFieldReferenceExpressions(@NotNull Data data) {
-    GoLiteralValue literalValue = data.getCompositeLit().getLiteralValue();
+  private static void moveFieldReferenceExpressions(@NotNull List<GoReferenceExpression> referenceExpressions,
+                                                    @Nullable GoLiteralValue literalValue,
+                                                    @NotNull GoAssignmentStatement parentAssignment) {
```

Comment:
what's the difference between `assigment` parameter which is used in almost all methods and `parentAssignment`